app/src/org/commcare/dalvik/activities/SingleAppManagerActivity.java

```diff
@@ -123,7 +123,11 @@ protected void onActivityResult(int requestCode, int resultCode, Intent intent)
         switch (requestCode) {
             case CommCareHomeActivity.UPGRADE_APP:
                 if (resultCode == RESULT_CANCELED) {
-                    Toast.makeText(this, R.string.update_canceled, Toast.LENGTH_LONG).show();
+                    UpdateTask task = UpdateTask.getRunningInstance();
```

Comment:
This is a bit awkward, timing wise, right? The instance could easily have started running again in a different thread by the time we're back here.